app/models/vm_migrate_workflow.rb

```diff
@@ -16,13 +16,16 @@ def self.default_dialog_file
   def get_source_and_targets(refresh = false)
     return @target_resource if @target_resource && refresh == false
 
-    ems = @values[:src_ids].to_miq_a.collect { |v_id| v = Vm.find_by_id(v_id); v.ext_management_system }.uniq.compact
+    vms = Vm.where(:id => @values[:src_ids].to_miq_a)
```

Comment:
You do not need the `to_miq_a` here.  The same call is being made below with out.